[Android] Allow setting the VideoLayout view to transparent color

quietvoid opened the pull request:
## Description
Add advanced setting to set the `VideoLayout` view background color on launch, for Android devices only.
Replaces PR https://github.com/xbmc/xbmc/pull/25171

Usage:
```
<advancedsettings version="1.0">
  <gui>
   <transparentvideolayout>true</transparentvideolayout>
 </gui>
</advancedsettings>
```

## Motivation and context
Sony TVs have a long standing firmware bug. Allow a user to configure the video layout background color to workaround this.
Fixes #25166 

## How has this been tested?
Runtime tested on FireTV Stick 4K Max which has the same issue but with transparent background color instead of black.
Needs testing by Sony TV users.

## What is the effect on users?
None by default. Requires advanced setting.

## Types of change
<!--- What type of change does your code introduce? Put an `x` with no space in all the boxes that apply like this: [X] -->
- [ ] **Bug fix** (non-breaking change which fixes an issue)
- [ ] **Clean up** (non-breaking change which removes non-working, unmaintained functionality)
- [x] **Improvement** (non-breaking change which improves existing functionality)
- [ ] **New feature** (non-breaking change which adds functionality)
- [ ] **Breaking change** (fix or feature that will cause existing functionality to change)
- [ ] **Cosmetic change** (non-breaking change that doesn't touch code)
- [ ] **Student submission** (PR was done for educational purposes and will be treated as such)
- [ ] **None of the above** (please explain below)

## Checklist:
<!--- Go over all the following points, and put an `X` with no space in all the boxes that apply like this: [X] -->
<!--- If you're unsure about any of these, don't hesitate to ask. We're here to help! -->
- [x] My code follows the **[Code Guidelines](https://github.com/xbmc/xbmc/blob/master/docs/CODE_GUIDELINES.md)** of this project 
- [x] My change requires a change to the documentation, either Doxygen or wiki
- [ ] I have updated the documentation accordingly
- [x] I have read the **[Contributing](https://github.com/xbmc/xbmc/blob/master/docs/CONTRIBUTING.md)** document
- [ ] I have added tests to cover my change
- [ ] All new and existing tests passed

fuzzard commented on xbmc/application/Application.cpp:
```diff
@@ -668,6 +668,13 @@ bool CApplication::Initialize()
       }
     }
 
+#if defined(TARGET_ANDROID)
+    if (CServiceBroker::GetSettingsComponent()->GetAdvancedSettings()->m_guiVideoLayoutTransparent)
+    {
+      CLog::Log(LOGINFO, "XBMCApp: VideoLayout view was set to transparent.");
+      CXBMCApp::Get().SetVideoLayoutBackgroundColor(0);
+    }
+#endif
```
is there a reason this cant be placed in one of the platform specific init stages, or in windowing?

quietvoid replied:
I'm not sure of the best place. I just reused fritsch's code.

quietvoid replied:
If anyone can clarify where the Android init stuff is, I can try it out.
Or other suggestions.

thexai replied:
Maybe move here:

```diff
diff --git a/xbmc/platform/android/activity/XBMCApp.cpp b/xbmc/platform/android/activity/XBMCApp.cpp
index 437c7371d9..64f4dff6ca 100644
--- a/xbmc/platform/android/activity/XBMCApp.cpp
+++ b/xbmc/platform/android/activity/XBMCApp.cpp
@@ -265,6 +265,12 @@ void CXBMCApp::onStart()
         std::make_unique<CJNIActivityManager>(getSystemService(CJNIContext::ACTIVITY_SERVICE));
     m_inputHandler.setDPI(GetDPI());
     runNativeOnUiThread(RegisterDisplayListenerCallback, nullptr);
+
+    if (CServiceBroker::GetSettingsComponent()->GetAdvancedSettings()->m_guiVideoLayoutTransparent)
+    {
+      CLog::Log(LOGINFO, "XBMCApp: VideoLayout view was set to transparent.");
+      SetVideoLayoutBackgroundColor(0);
+    }
   }
 }
```
 
not tested although

quietvoid replied:
I've tried moving it but `CServiceBroker::GetSettingsComponent()` ends up being a nullptr at that stage.

CastagnaIT replied:
you can try move it on `CPlatformAndroid` (PlatformAndroid.cpp)
by add a new override to use `InitStageThree` method

quietvoid replied:
Thanks. That works. :+1:

quietvoid committed:
[Android] Allow setting the VideoLayout view to transparent color

Sony TVs have a long standing firmware bug. Allow a user to configure
the video layout background color to workaround this.
Fixes #25166

Usage:
```
<advancedsettings version="1.0">
  <gui>
    <transparentvideolayout>true</transparentvideolayout>
  </gui>
</advancedsettings>
```

thexai reviewed (approved):
Changes looks good to me.

thexai commented:
Jenkins build this please

thexai commented:
Jenkins build this please

thexai commented:
jenkins build this please

quietvoid commented:
Does this still need to be backported to Nexus?

thexai commented:
Yes, I think it is necessary. Thanks.
